web/src/topic_list_data.js

```diff
@@ -120,6 +120,11 @@ export function get_list_info(stream_id, zoomed, search_term) {
     let topic_names = stream_topic_history.get_recent_topic_names(stream_id);
     if (zoomed) {
         topic_names = util.filter_by_word_prefix_match(topic_names, search_term, (item) => item);
+        if (topics_state !== "all") {
+            topic_names = topic_names.filter(
+                (name) => resolved_topic.is_resolved(name) === (topics_state === "resolved"),
```

Comment:
Let's extract `(topics_state === "resolved")` into a variable so that it is easier to read and we don't have do that calculation in a loop.

Reply:
Done